### PR TITLE
Add allowDigitGroupSeparators to numericality

### LINKED ISSUE #225: Numericality validation not compatible with Ember data 'number' type
[Ember Data uses `Number(value)` to serialize attributes](https://github.com/emberjs/data/blob/v1.0.0-beta.14.1/packages/ember-data/lib/transforms/number.js#L49) whereas [Ember Validations matches against a pattern which cleverly allows commas](https://github.com/dockyard/ember-validations/blob/master/addon/patterns.js#L4).

```javascript
var value = '123,456'; // valid via the numericality matcher

// Ember Data will serialize it like so…
Number('123,456') // NaN
```

The resulting behavior is that a field will appear as valid, however upon saving that value is converted to `null`.

I appreciate that those fancy commas are sometimes desirable, however I'm not sure they are a sensible default behavior. Perhaps they could be enabled via a boolean toggle option?

### PR DESCRIPTION
**Pull request by soulcutter**
Alters the default behavior of the numericality validator to more
closely-match what can be parsed by `Number(value)`.

The previous behavior - which allowed numbers such as 1,234 - can be
re-enabled via passing `allowDigitGroupSeparators: true` as an option.

For completeness and internationalization this also supports passing a
character to use as the separator, `allowDigitGroupSeparators: '.'` as well
as adding a `decimalMark: ','` option (default is `'.'`). Used together this
adds support for validating european-style currency numbers such as
1.234.567,89 successfully.

Fixes #225

### CODE REVIEW

**Comment by soulcutter** [comment]
FWIW that CI failure has nothing to do with this PR - if you remove the commented-out remote uniqueness validator file, this should pass.

**Comment by soulcutter** [comment]
Having given this a bit more thought, I'm wondering if the numericality validator should even contain the option to accept those commas at all. It adds complexity, and is not an international standard - http://www.statisticalconsultants.co.nz/blog/how-the-world-separates-its-digits.html 

The only thing that would be a shame about removing it is that most people would struggle to come up with that regex… but I don't know that's a good enough reason to keep supporting it.

Waiting for feedback before making any further changes.

**Comment by soulcutter** [comment]
Welp, I lied about waiting for feedback. I went ahead and made the digit group separator character configurable, and added a `decimalMark` option as well. I think that about covers most number formats.

I'm personally still on the fence a bit on whether or not that sort of formatting is something that should be tackled in this lib/validator. For one thing it's pretty well guaranteed not to work with some of the other `CHECKS` which depend on javascript's `parseInt` and whatnot. That's a situation that already exists today with supporting `'123,456'` out of the box, though, so I think this is no worse.

**Comment by bcardarella** [comment]
@soulcutter the reason why the comma is being tested for was due to `input` element's forcing the type being set on the property to `string`. While Ember Data will serialize this type back to numerical, if you have validations outside of an ember data model you want to cover this type of behavior.

**Comment by soulcutter** [comment]
@bcardarella I think this PR aligns with what you're saying. To be clear, you do want to support commas separating groups of digits?

This PR makes that validation more configurable and changes the default behavior to not allow commas. I think changing the default may be the more-controversial bit, however I believe it's justified because it more closely aligns with what JavaScript can parse as a number. Also consider that it improves the default behavior of checks such as `> 0` - currently a value of `1,234` will fail validation saying that the number must be greater than zero.

Having said that, this PR enables you to explicitly allow the old behavior, and actually improves it insofar that you may validate European-style currency numbers (where the digit group separator is a period, and the decimal mark is a comma). Another improvement is the ability to allow digit group separators for `integerOnly` Validation.

TLDR; you can do everything same as before and more, but the default is perhaps less-surprising.

**Comment by bcardarella** [comment]
@soulcutter would it be OK if this sits for a bit? I plan on spending February in OSS land when I can take this into real consideration. I'd like to go back and revisit why we allowed the comma and vet that against what you're saying

**Comment by soulcutter** [comment]
Yeah, that's ok. Thanks for communicating your thoughts on a timeline, I can put this out of my mind for a while.

In case anyone stumbles upon this in the meantime, the easy workarounds are:

* Add a format validation that catches commas that would otherwise pass through numericality checks
* Customize the `number` transform ala `app/transforms/number.js` like so:

```javascript
import DS from 'ember-data';

export default DS.NumberTransform.extend({
  serialize: function(value) {
    if (value) {
      // sanitize numbers such as 1,234 to 1234
      value = String(value).replace(',', '');
    }
    return this._super(value);
  }
});
```

**Comment by martndemus** [comment]
I am going to close this PR for now, feel free to reopen this PR if you feel like it's still relevant and we should take a look at it. If you do, please make sure the code is rebased against master.
